packages/x-data-grid/src/internals/utils/propValidation.ts

```diff
@@ -13,6 +14,11 @@ export const propValidatorsDataGrid: PropValidator<DataGridProcessedProps>[] = [
         'Please remove one of these two props.',
       ].join('\n')) ||
     undefined,
+  (props) =>
```

Comment:
I think we should add a test case for this one, like the case above has.

Reply:
Done in [`7079c39` (#12448)](https://github.com/mui/mui-x/pull/12448/commits/7079c39d8cd4b1973deafa1bf3e68267119a4c08)